pkg/operator/apis/monitoring/v1/http_types.go

```diff
@@ -24,145 +24,119 @@ import (
-	if s.Secret == nil {
-		return nil
+// SecretConfigs returns unordered list of secrets.SecretConfig[kubernetes.SecretConfig].
+func (p PrometheusSecretConfigs) SecretConfigs() (ret []secrets.SecretConfig[kubernetes.SecretConfig]) {
+	ret = make([]secrets.SecretConfig[kubernetes.SecretConfig], 0, len(p))
+	for ref, c := range p {
+		ret = append(ret, secrets.SecretConfig[kubernetes.SecretConfig]{
+			Name:   ref,
+			Config: c,
+		})
 	}
-	secret := s.Secret.GetClusterSecret()
-	return &secret
+	return ret
 }
 
-type ClusterSecretKeySelector struct {
-	// The secret name and key to retrieve (inlined).
-	SecretKeySelector `json:",inline"`
-
-	// The namespace to retrieve the secret from.
-	Namespace string `json:"namespace"`
-}
+// toPrometheusSecretRef returns Prometheus reference to Kubernetes secret (or empty string if not set).
+// It also adds secret to list of secrets by argument reference, if exists.
+// It returns error if namespace tenancy security restriction is violated for PodMonitoring.
+func (s *ClusterSecretKeySelector) toPrometheusSecretRef(m PodMonitoringCRD, pool PrometheusSecretConfigs) (string, error) {
+	if s == nil {
+		return "", nil
+	}
 
-func (s *ClusterSecretKeySelector) GetClusterSecretKey() string {
-	return s.SecretKeySelector.GetLocalSecretKey(s.Namespace)
-}
+	ns := s.Namespace
+	if ns == "" {
+		ns = m.GetNamespace()
+	} else if m.IsNamespaceScoped() && ns != m.GetNamespace() {
+		return "", fmt.Errorf("PodMonitoring secret selector can't select secret from the different namespace than PodMonitoring namespace %q, got: %q; Consider using ClusterPodMonitoring or copy secret to PodMonitoring namespace", m.GetNamespace(), ns)
+	}
 
-func (s *ClusterSecretKeySelector) GetClusterSecret() secrets.SecretConfig[kubernetes.SecretConfig] {
-	return s.SecretKeySelector.GetLocalSecret(s.Namespace)
+	ref := fmt.Sprintf("%s/%s/%s", ns, s.Name, s.Key)
+	pool.Set(ref, kubernetes.SecretConfig{
+		Namespace: ns,
+		Name:      s.Name,
+		Key:       s.Key,
+	})
+	return ref, nil
 }
 
-// Auth sets the `Authorization` header on every scrape request.
-//
-// Currently the credentials are not configurable and always empty.
+// Auth sets the `Authorization` header on every HTTP request.
 type Auth struct {
-	// The authentication type. Defaults to Bearer, Basic will cause an error.
-	Type        string                `json:"type,omitempty"`
-	Credentials ClusterSecretSelector `json:"credentials,omitempty"`
+	// The authentication type. Defaults to Bearer.
+	// Basic will cause an error, as the BasicAuth object should be used instead.
+	Type string `json:"type,omitempty"`
+	// CredentialsSecret references the Kubernetes secret's key with the credentials
+	// (token) for the auth header to send along the request.
+	// Optional, as in previous resource versions we allowed no credentials.
+	// +optional
+	CredentialsSecret *ClusterSecretKeySelector `json:"credentialsSecret,omitempty"`
```

Comment:
suggestion: Does flattening them make it any easier for users? An extra indentation should not be much difficult.

The biggest benefit for splitting it into a separate type is the flexibility of adding additional secret types, e.g. ConfigMaps or Google Cloud Secret Manager. Then, there's also compatibility with the existing `Authorization` object which we should eventually merge.

Thoughts?